frame/collective/src/benchmarking.rs

```diff
@@ -127,152 +150,161 @@ benchmarks_instance! {
+		let b = MAX_BYTES;
 
 		// Construct `members`.
 		let mut members = vec![];
-		for i in 0 .. m {
+		let proposer: T::AccountId = account("proposer", 0, SEED);
+		members.push(proposer.clone());
+		for i in 1 .. m - 1 {
 			let member = account("member", i, SEED);
 			members.push(member);
 		}
-
-		let caller: T::AccountId = account("caller", 0, SEED);
-		members.push(caller.clone());
+		let voter: T::AccountId = account("voter", 0, SEED);
+		members.push(voter.clone());
 		Collective::<T, _>::set_members(SystemOrigin::Root.into(), members.clone(), None)?;
 
 		// Threshold is 1 less than the number of members so that one person can vote nay
-		let threshold = m;
+		let threshold = m - 1;
 
 		// Add previous proposals
 		let mut last_hash = T::Hash::default();
 		for i in 0 .. p {
 			// Proposals should be different so that different proposal hashes are generated
 			let proposal: T::Proposal = frame_system::Call::<T>::remark(vec![i as u8; b as usize]).into();
-			Collective::<T, _>::propose(SystemOrigin::Signed(caller.clone()).into(), threshold, Box::new(proposal.clone()))?;
+			Collective::<T, _>::propose(SystemOrigin::Signed(proposer.clone()).into(), threshold, Box::new(proposal.clone()))?;
 			last_hash = T::Hashing::hash_of(&proposal);
 		}
 
-		// Have everyone vote aye on last proposal, while keeping it from passing
-		for j in 2 .. m {
+		let index = p - 1;
+		// Have most everyone vote aye on last proposal, while keeping it from passing.
+		// Proposer already voted aye so we start at 1.
+		for j in 1 .. m - 3 {
 			let voter = &members[j as usize];
 			let approve = true;
-			Collective::<T, _>::vote(SystemOrigin::Signed(voter.clone()).into(), last_hash.clone(), p - 1, approve)?;
+			Collective::<T, _>::vote(SystemOrigin::Signed(voter.clone()).into(), last_hash.clone(), index, approve)?;
 		}
+		// Voter votes aye without resolving the vote.
+		let approve = true;
+		Collective::<T, _>::vote(SystemOrigin::Signed(voter.clone()).into(), last_hash.clone(), index, approve)?;
 
 		assert_eq!(Collective::<T, _>::proposals().len(), p as usize);
 
-		// Caller switches vote to nay, but does not kill the vote, just updates + inserts
-		let index = p - 1;
+		// Voter switches vote to nay, but does not kill the vote, just updates + inserts
 		let approve = false;
 
-	}: vote(SystemOrigin::Signed(caller), last_hash.clone(), index, approve)
+	}: _(SystemOrigin::Signed(voter), last_hash.clone(), index, approve)
 	verify {
 		// All proposals exist and the last proposal has just been updated.
 		assert_eq!(Collective::<T, _>::proposals().len(), p as usize);
 		let voting = Collective::<T, _>::voting(&last_hash).ok_or(Error::<T, I>::ProposalMissing)?;
-		assert_eq!(voting.ayes.len(), (m - 2) as usize);
+		assert_eq!(voting.ayes.len(), (m - 3) as usize);
 		assert_eq!(voting.nays.len(), 1);
 	}
 
-	vote_disapproved {
-		let m in 2 .. MAX_MEMBERS;
-		let p in 1 .. MAX_PROPOSALS;
+	close_early_disapproved {
+		let m in 5 .. T::MaxMembers::get();
```

Comment:
Why 5?

Reply:
The different members have different "roles" (approve, disapprove, propose, whatever) and 5 makes sure that all can be fulfilled. (So that ranges like `2..m-2` are meaningful.)
I should probably document this more :sweat_smile: